File: Pyg_shop/interface/src/main/java/top/takefly/core/service/BrandService.java
package top.takefly.core.service;

import top.takefly.core.pojo.entity.PageResult;
import top.takefly.core.pojo.entity.Result;
import top.takefly.core.pojo.good.Brand;

import java.util.List;
import java.util.Map;

public interface BrandService {

    /**
     * 根据id查询品牌信息
     * @param id
     * @return
     */
    public Brand findBrandById(Long id);

    /**
     * 查询所有品牌信息
     * @return
     */
    public List<Brand> findAll();

    /**
     * 分页显示品牌信息
     * @param page
     * @param size
     * @return
     */
    public PageResult findPage(Integer page, Integer size);

    /**
     * 保存品牌信息
     * @param brand
     */
    public void save(Brand brand);

    public Brand findOne(Long id);

    /**
     * 修改品牌信息
     * @param brand
     */
    public void update(Brand brand);

    public void delDetach(Long[] ids);

    public PageResult searchBrand(Brand brand, Integer page, Integer size);
    /**
     * 响应模板的下拉列表的数据支撑
     *
     * @return
     */
    public List<Map> selectBrandToTypeTemplate();
}

File: Pyg_shop/web_manager/src/main/webapp/js/service/brandService.js
app.service('brandService', function ($http) {
    //分页
    this.findPage = function (page, size) {
        return $http.get('../brand/findPage.do?page=' + page + "&size=" + size);
    }

    //更新操作
    this.update = function (entity) {
        return $http.post('../brand/update.do', entity);
    }

    this.save = function (entity) {
        return $http.post('../brand/save.do', entity);
    }

    //数据回显
    this.findOne = function (id) {
        return $http.get('../brand/findOne.do?id=' + id);
    }

    //批量删除
    this.delDetach = function (ids) {
        return $http.get('../brand/delDetach.do?ids=' + ids);
    }

    //搜索品牌
    this.search = function (page, size, brandentity) {
        return $http.post('../brand/search.do?page=' + page + '&size=' + size, brandentity);
    }
})